helix-term/src/ui/editor.rs

```diff
@@ -830,6 +833,53 @@ impl EditorView {
         }
     }
 
+    /// Apply the highlighting on the columns where a cursor is active
+    pub fn highlight_cursorcolumn(
+        doc: &Document,
+        view: &View,
+        surface: &mut Surface,
+        theme: &Theme,
+    ) {
+        let text = doc.text().slice(..);
+
+        // Manual fallback behaviour:
+        // ui.cursorcolumn.{p/s} -> ui.cursorcolumn -> ui.cursorline.{p/s}
+        let primary_style = theme
+            .try_get_exact("ui.cursorcolumn.primary")
+            .or_else(|| theme.try_get_exact("ui.cursorcolumn"))
+            .unwrap_or_else(|| theme.get("ui.cursorline.primary"));
+        let secondary_style = theme
+            .try_get_exact("ui.cursorcolumn.secondary")
+            .or_else(|| theme.try_get_exact("ui.cursorcolumn"))
+            .unwrap_or_else(|| theme.get("ui.cursorline.secondary"));
```

Comment:
This could actually be achieved with the existing `try_get` without introducing `try_get_exact`:

```rust
        let primary_style = theme
            // checks `ui.cursorcolumn.primary` first, then `ui.cursorcolumn`
            .try_get("ui.cursorcolumn.primary")
            // if `ui.cursorcolumn` isn't defined, falls back to 
            // `ui.cursorcolumn.primary` and then to `ui.cursorcolumn`
            .unwrap_or_else(|| theme.get("ui.cursorline.primary"));
        let secondary_style = theme
            .try_get("ui.cursorcolumn.secondary")
            .unwrap_or_else(|| theme.get("ui.cursorline.secondary"));
```